### PR TITLE
Remove dependency of class collection

### LINKED ISSUE #44: Remove Dependency of Classes on Collection
### I'm submitting a 
- [ ] bug report.
- [x] feature request.

### Current Behaviour:

Presently, class and a collection are tightly coupled. It is bound that a collection will contain members of same class, while the spec defines collections as somehow related resources. It is possible to have instances of different classes as the members of the same collection. 

### Expected Behaviour:

It would be nice to allow API authors to decide about the members of the class by passing manages block. There should not be any restriction of passing a class to make a collection in the doc_writer module. 

### Do you want to work on this issue?

yes

### PR DESCRIPTION
**Pull request by priyanshunayan**
Fixes #44 

### Checklist
- [x] My branch is up-to-date with upstream/develop branch.
- [x] Everything works and tested for Python 3.5.2 and above.

### Description

This PR removes the dependency of collections on classes. Earlier, one had to use `add_supported_class(classname, collection=True)`
Now the writers have flexibility to tweak collections as per their preferences. They can create collections and then add to HydraDoc via ` apidoc.add_supported_collection(collection_name)`
In other words, now there is no forced dependency of classes on collection

### Change logs

- in the doc_writer_sample.py, passed full URI instead of passing one word URIs
- added method to add collections to Hydradoc
- some compact URIs were not expanding when used expansion algorithm was used, for eg: `_:singleclass_retrieve`. Replaced them with proper URIs.

### CODE REVIEW

File: samples/doc_writer_sample_output.py [hunk 1]
```diff
@@ -487,7 +499,7 @@
                                 "@id": "updateclass",
                                 "@type": "http://schema.org/UpdateAction",
                                 "description": "null",
-                                "expects": "vocab:singleClass",
+                                "expects": "https://hydrus.com/api/singleClass",
```
Comment (chrizandr) [inline samples/doc_writer_sample_output.py]: We don't use the endpoint URI here, this should lead to the definition of what it expects. This would be 'vocab:singleClass'.

File: samples/doc_writer_sample_output.py [hunk 2]
```diff
@@ -347,112 +359,112 @@
...
                     "expectsHeader": [],
                     "method": "GET",
                     "possibleStatus": [],
-                    "returns": "vocab:extraClassCollection",
+                    "returns": "https://hydrus.com/api/dummyClass",
                     "returnsHeader": []
                 },
                 {
-                    "@id": "_:extraclass_create",
+                    "@id": "https://hydrus.com/api/collection_2/dummyclasses_create",
```
Comment (chrizandr) [inline samples/doc_writer_sample_output.py]: This is wrong. https://hydrus.com/api/collection_2/dummyclasses_create and https://hydrus.com/api/collection_2:dummyclasses_create are very different. When we use `/` instead of `:` we change the URI. Adding `:` is used to define a namespace, the URL would still be https://hydrus.com/api/collection_2

File: samples/doc_writer_sample_output.py [hunk 3]
```diff
@@ -560,7 +572,7 @@
                                         "title": ""
                                     }
                                 ],
-                                "returns": "vocab:singleClass",
+                                "returns": "https://hydrus.com/api/singleClass",
```
Comment (chrizandr) [inline samples/doc_writer_sample_output.py]: Same for returns, it would be vocab:singleClass. We need the definition, there is no definition on https://hydrus.com/api/singleClass, that is for the instance

File: samples/doc_writer_sample_output.py [hunk 4]
```diff
@@ -276,7 +288,7 @@
                 },
                 {
                     "@type": "SupportedProperty",
-                    "property": "vocab:anotherSingleClass",
+                    "property": "https://hydrus.com/api/anotherSingleClass",
```
Comment (chrizandr) [inline samples/doc_writer_sample_output.py]: It's fine to expand this here because this is for doc_writer, but make sure that we are able to change this when we parse the same document with doc_maker and we change the URI and API name in that. Not sure how you can ensure that. I would suggest not giving outputs in expanded form, and instead use expanded forms only internally in python.

**Review by chrizandr — CHANGES_REQUESTED** [review]

**Comment by priyanshunayan** [comment]
@chrizandr I have made required changes. Please take a look.

File: samples/doc_writer_sample_output.py [hunk 5]
```diff
@@ -413,31 +450,31 @@
...
             "manages": {
-                "object": "https://hydrus.com/api/dummyClass",
+                "object": "https://hydrus.com/api/vocab#dummyClass",
                 "property": "rdf:type"
             },
             "subClassOf": "http://www.w3.org/ns/hydra/core#Collection",
             "supportedOperation": [
                 {
-                    "@id": "https://hydrus.com/api/collection_2/dummyclasses_retrieve",
+                    "@id": "https://hydrus.com/api/vocab#dummyclasses_retrieve",
```
Comment (chrizandr) [inline samples/doc_writer_sample_output.py]: This is again wrong. The @id here needs to be https://hydrus.com/api/collection_2:dummyclasses_retrieve. Here the @id refers to the endpoint where the operation is done, not the definition of the operation. If endpoint is not known we use the _:dummyclasses_retrieve convention.
